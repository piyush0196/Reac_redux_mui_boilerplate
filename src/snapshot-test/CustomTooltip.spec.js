import TestRenderer from 'react-test-renderer'
import CustomTooltip from 'components/ui-kit/Tooltip'

describe('Component - CustomTooltip', () => {
  const wrappers = []

  const ToolTipJSX = ({ type, title, arrow, placement }) => {
    return (
      <CustomTooltip type={type} title={title} arrow={arrow} placement={placement}>
        <div style={{ backgroundColor: 'lightgray', width: '150px' }}>
          Hover over me
        </div>
      </CustomTooltip>
    )
  }

  wrappers.push(TestRenderer.create(
    <ToolTipJSX title='test label' placement='bottom-end' />
  ))
  wrappers.push(TestRenderer.create(
    <ToolTipJSX title='test label' arrow placement='top' />
  ))
  wrappers.push(TestRenderer.create(
    <ToolTipJSX title='test label' type='error' placement='left' />
  ))

  it('renders all CustomTooltip correctly', () => {
    wrappers.forEach(wrapper => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})
