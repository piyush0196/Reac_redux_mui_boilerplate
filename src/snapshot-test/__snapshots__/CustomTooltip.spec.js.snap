// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - CustomTooltip renders all CustomTooltip correctly 1`] = `
<div
  aria-label="test label"
  aria-labelledby={null}
  className=""
  data-mui-internal-clone-element={true}
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={
    Object {
      "backgroundColor": "lightgray",
      "width": "150px",
    }
  }
  type="default"
>
  Hover over me
</div>
`;

exports[`Component - CustomTooltip renders all CustomTooltip correctly 2`] = `
<div
  aria-label="test label"
  aria-labelledby={null}
  className=""
  data-mui-internal-clone-element={true}
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={
    Object {
      "backgroundColor": "lightgray",
      "width": "150px",
    }
  }
  type="default"
>
  Hover over me
</div>
`;

exports[`Component - CustomTooltip renders all CustomTooltip correctly 3`] = `
<div
  aria-label="test label"
  aria-labelledby={null}
  className=""
  data-mui-internal-clone-element={true}
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={
    Object {
      "backgroundColor": "lightgray",
      "width": "150px",
    }
  }
  type="error"
>
  Hover over me
</div>
`;
